src/attributes.js

```diff
@@ -29,7 +29,7 @@ var applyAttr = function(el, name, value) {
   if (value == null) {
     el.removeAttribute(name);
   } else {
-    el.setAttribute(name, value);
+    el.setAttribute(name, /**@type {boolean|number|string}*/(value));
```

Comment:
So your right that the inline type should be moved up to the param for `applyAttr`. There is a problem with `applyAttributeTyped` then. I'm not really sure why the compiler is not smart enought to infer this tho without the inline type.
```js
  var type = typeof value;
  if (type === 'object' || type === 'function') {
    applyProp(el, name, value);
  } else {
    applyAttr(el, name, /** @type {?(boolean|number|string)} */(value));
  }
```